var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -229,7 +229,7 @@ def do_stage(self, mirror_only=False):
     def nmake_arguments(self):
         args = []
         if self.spec.satisfies("%msvc"):
-            args.append("CCTYPE=%s" % self.compiler.msvc_version)
+            args.append("CCTYPE=%s" % self.compiler.short_msvc_version)
```

Comment:
Note if you create a version object, it will automatically create separate indexable components based on `.`'s, so IMO it would be more idiomatic to set `msvc.version = Version(...)` and then you can say here `self.compiler.msvc_version[:2]` (or however many components that `perl` wants).

Reply:
A common use case that a number of nmake packages require is to prefix the first three numbers in the version string with `MSVC` such as `MSVC142`. It's a common enough case I think the `Msvc` compiler class should provide the method. That being said I did not know the version object could do that, so I'll refactor the MSVC side to use that functionality. Super useful the version object can do that.